module/validation/receipt_validator_test.go

```diff
@@ -224,6 +224,28 @@ func (s *ReceiptValidationSuite) TestReceiptForBlockWith0Collections() {
 	})
 }
 
+// TestReceiptInconsistentChunkList tests that we reject receipts when the Start and End states
+// within the chunk list are inconsistent (e.g. chunk[0].EndState != chunk[1].StartState).
+func (s *ReceiptValidationSuite) TestReceiptInconsistentChunkList() {
+	s.publicKey.On("Verify", mock.Anything, mock.Anything, mock.Anything).Return(true, nil).Maybe()
+	valSubgrph := s.ValidSubgraphFixture()
+	chunks := valSubgrph.Result.Chunks
+	require.GreaterOrEqual(s.T(), chunks.Len(), 1)
+	// swap last chunk's start and end states
+	lastChunk := chunks[len(chunks)-1]
+	tmp := lastChunk.StartState
+	lastChunk.StartState = lastChunk.EndState
+	lastChunk.EndState = tmp
```

Comment:
```suggestion
	lastChunk.StartState, lastChunk.EndState = lastChunk.EndState, lastChunk.StartState
```
We can do this inline as well